recipes/h5pp/all/conanfile.py

```diff
@@ -66,6 +66,11 @@ def package_info(self):
         self.cpp_info.components["h5pp_deps"].set_property("cmake_target_name", "h5pp::deps")
         self.cpp_info.components["h5pp_deps"].requires = ["eigen::eigen", "spdlog::spdlog", "hdf5::hdf5"]
         self.cpp_info.components["h5pp_flags"].set_property("cmake_target_name", "h5pp::flags")
+        if tools.Version(self.version) >= "1.10.0":
+            self.cpp_info.components["h5pp_flags"].defines.append("H5PP_USE_EIGEN3")
+            self.cpp_info.components["h5pp_flags"].defines.append("H5PP_USE_SPDLOG")
+            self.cpp_info.components["h5pp_flags"].defines.append("H5PP_USE_FMT")
```

Comment:
Will this list of `defines` be overriden by the line below with `=` in it?

```python
        if is_msvc(self):
            self.cpp_info.components["h5pp_flags"].defines = ["NOMINMAX"]
```

Reply:
> Will this list of `defines` be overriden by the line below with `=` in it?
> 
> ```python
>         if is_msvc(self):
>             self.cpp_info.components["h5pp_flags"].defines = ["NOMINMAX"]
> ```

You are correct, good catch! It should be appended.

> Besides the commented details, I also think this requirements and flags should depend on some option so consumers can choose to include them or not. Of course, this is not needed for this PR, but if you want to add them as well, that would be very welcome.
> 
> Thanks!

That's a fair point. I'll try adding options like "with_<lib>"  that default to True, since that is the nominal state of h5pp. But then users can opt out if needed. I'd prefer adding it to this PR if that's ok.

Reply:
These changes are now added in [26717f4](https://github.com/conan-io/conan-center-index/pull/10697/commits/26717f46a553ba545b1fe0e75eb5f8f649c69721) and [a9d1e2f](https://github.com/conan-io/conan-center-index/pull/10697/commits/a9d1e2f9208c8f98d60bee2ba9cfe901fc816bee), but the change request is still marked as pending. I don't know how to fix that